sds_data_manager/lambda_code/IAlirtCode/ialirt_ingest.py

```diff
@@ -27,23 +28,43 @@ def lambda_handler(event, context):
         and runtime environment.
 
     """
+    # TODO: these steps will be put into different functions.
     logger.info("Received event: %s", json.dumps(event))
 
-    table_name = os.environ.get("TABLE_NAME")
+    ingest_table_name = os.environ.get("INGEST_TABLE")
+    algorithm_table_name = os.environ.get("ALGORITHM_TABLE")
     dynamodb = boto3.resource("dynamodb")
-    table = dynamodb.Table(table_name)
+    ingest_table = dynamodb.Table(ingest_table_name)
+    algorithm_table = dynamodb.Table(algorithm_table_name)
 
     s3_filepath = event["detail"]["object"]["key"]
     filename = os.path.basename(s3_filepath)
     logger.info("Retrieved filename: %s", filename)
 
-    # TODO: item is temporary and will be replaced with actual packet data.
+    # TODO: Each of these steps in temporary, but provides an idea
+    #  of how the lambda will be used.
+    # 1. Ingest Data to Ingest Table.
     item = {
         "apid": 478,
         "met": 123,
         "ingest_time": "2021-01-01T00:00:00Z",
         "packet_blob": b"binary_data_string",
     }
 
-    table.put_item(Item=item)
+    ingest_table.put_item(Item=item)
+    logger.info("Successfully wrote item to DynamoDB: %s", item)
+
+    # 2. Query Ingest Table for previous times as required by instrument.
+    response = ingest_table.query(KeyConditionExpression=Key("apid").eq(478))
+    items = response["Items"]
+    logger.info("Scan successful. Retrieved items: %s", items)
+
+    # 3. After processing insert data into Algorithm Table.
```

Comment:
Will processing occur inside this lambda as well? What step would that be?

Reply:
Yes. I will import the processing code for each instrument from imap_processing. The full dataset is stored in different packets. So what I will do is:

1. query L0 DB for the past 5 min
2. xarray_data = packet_file_to_datasets(binary_packet_path, xtce_hit_path)[apid]
3. hit_product = process_hit(xarray_data)
4. insert hit_product (which is a dictionary) into final DynamoDB

Caveat: I know that this might result in duplicate data products trying to be inserted into the final DynamoDB. I haven't quite decided what to do about that yet, but it is for another PR I think.